xknx/config/config_v1.py

```diff
@@ -59,6 +62,17 @@ def parse_general(self, doc):
             if "multicast_port" in doc["general"]:
                 self.xknx.multicast_port = doc["general"]["multicast_port"]
 
+    def env_general(self):
```

Comment:
Hint: This is the old config format. We are soon(tm) be migrating to xknx schema v2 - where I will just remove this file in its entirety.

I like the idea of having environment variables in place but I do not agree on the approach taken. I'd prefer having this in the yaml and introduce a dedicated yaml loader for env vars. I can take care of this in the new config schema.

Reply:
Ok, I will use the new config file.

The practice of loading env after the config file allow us to setup all the stable configuration in the file (devices for ex.), and overwrite all the context dependent attributes from env vars that are set during the container deployment (IPs, ports, etc.). This is a common way of working in container's clusters like Kubernetes and Docker Swarm (I'm using the second one for home automation).

Reply:
I would still prefer giving the user the ability to change the env vars and also add new ones depending on their needs.

I'm using kubernetes for my home assistant and HA also supports this kind of syntax:

`local_port: !env_var XKNX_LOCAL_PORT`

Something like this. And then you can just use this in your docker compose/deployment.yaml

I prepare a PR for this tomorrow as most of it is already done. I like the other changes though.

Reply:
See https://github.com/XKNX/xknx/pull/480